[#32] feat : create register location

kagrin97 opened the pull request:
# 체크리스트

- [x] 작업 내용이 스프린드 보드에서 할당한 내용과 동일합니다.
- [x] 커밋 메시지가 가이드라인을 따릅니다.
- [x] 코딩 컨벤션 가이드라인을 준수하였습니다.
- [ ] 변경 사항에 대한 테스트가 추가되었습니다.
- [x] 기존의 모든 자동화된 테스트를 통과합니다.

# 설명

- script로 kakao sdk를 다운로드합니다.
- kakao 객체의 타입을 명시한 kakao.maps.d.ts 패키지를 설치 및 적용
- useGeoLocation으로 유저의 위도,경도를 받아옴
- useFindNeighborhoods에 위도,경도 props로 보내 사용자의 동네와 근처 동네들을 가져옵니다.
- keywordSearch메소드의 callback함수의 result props의 타입을 kakao.maps.d.ts에서 import할수가 없어서 any를 임시로 사용했습니다.


# 변경 사항

해당하는 작업에 대한 변경 사항을 상세하게 설명해주세요.

# 관련 이슈

#34

# 테스트 방법

useGeoLocation은 해당 유저의 위도, 경도를 받는 훅으로 테스트 코드를 작성하는 의미가 없다고 생각했습니다.

useFindNeighborhoods은 kakao map sdk를 다운로드해 kakao 객체를 사용하는 훅이라서 test코드를 작성하는데 많은 문제가
발생하였고 외부 리소스에 의존하는 코드라고 생각해 test코드를 작성하지 않았습니다.

kagrin97 committed:
[#32] feat : Applying Kakao MAP API SDK

	kakao map sdk를 index.html에 script로 불러옵니다.
	또한 kakao.maps.d.ts 패키지를 불러와 kakao obj의 타입을 명시합니다.

kagrin97 committed:
[#32] feat : ADD useGeoLocation hook

	사용자의 위도,경도를 얻는 훅을 작성했습니다

kagrin97 committed:
[#32] feat : ADD neighborhood finder hook

	사용자한테 위도,경도를 받고 해당정보로 주변 동네들을 가져오는 hook을 작성

kagrin97 committed:
[#32] refactor : Replace kakao map api with kakao local api

	기존의 kakao map sdk를 다운로드해서 사용하는 방식에서
	local api로 직접 axios 요청을 보내는 방식으로 바꿨습니다.

gomgun-lab commented on src/hooks/useGeoLocation.ts:
useGeoLocation 훅은 로케이션 정보를 가져오는데 유저의 권한에 따라 에러 처리를 해주면 좋을 것 같아요, 예를 들어서 유저가 권한을 설정하지 않았으면 에러 처리를 해서 그에 대한 UI 피드백을 줄 수 있을 것 같습니다. (비동기 에러 처리와 비슷하게)?

아래는 한번 참고하시면 좋을 것 같습니다.

- https://usehooks.com/usegeolocation
- https://github.com/uidotdev/usehooks/blob/main/index.js

또 Coordinates 타입을 kakaoMap.type 파일에서 가져와서 useGeoLocation 훅이 해당 도메인(kakao)에 직접 의존하는 것 같아요, 이보다는 Coordinates 타입을 따로 분리해서 useGeoLocation 훅이 특정 도메인에 의존하지 않아서 좀 더 범용적으로 useGeoLocation 사용할 수 있지 않을까 합니다.

민규님 의견은 어떠신가요?

p.s - 타입과 관련한 파일의 확장자를 type.ts 라고 명시하는 건 의도하신 건가요? 이 부분도 나중에 컨벤션 관련해서 얘기해보면 좋을 듯 하네요 ㅋㅋ

kagrin97 replied:
useGeoLocation 동작중에 에러가 발생할경우 isError Boolean 변수를 넘겨주는 방법도 좋을것 같네요.

기능을 정신없이 구현하다보니까 Coordinates 타입을 kakaoMap.type에 저장을 했네요;; 
말씀대로 따로 분리해야겠네요.

type.ts 라고 한건 의도한거죠 .component, .styles, .type 이렇게 최대한 하는것이 좋을(?) 것 같아요

kagrin97 committed:
[#32] refactor : Move the Coordinates interface to the location file

	기존의 kakaoMap.type.ts에 존재하던 Coordinates interface를
	location.type.ts파일로 옮겼습니다.

kagrin97 committed:
[#32] feat : Return additional isLoadingGeoLocation and errorGeoLocation values from useGeoLocation

	useGeoLocation에서 isLoadingGeoLocation errorGeoLocation값을 추가로 리턴
	isLoadingGeoLocation은 로딩상태를 Booelan으로 리턴
	errorGeoLocation 객체는 isError라는 Boolean값과
	errorMessage 문자열을 리턴

kagrin97 committed:
[#32] refactor : Remove unnecessary keywords from filenames

	파일이름에 꼭 필요하지 않는 키워드를 삭제했습니다
